File: www/website/cgi-bin/video.pl
use strict;
use warnings;

my $video_path = "/home/aismaili/Webserv/website/pics/NPC.mp4";

# Check if the video file exists
unless (-e $video_path) {
    print "Content-Type: text/plain\n\n";
    print "Error: Video file not found.";
    exit 1;
}

# Get the size of the video file
my $video_size = -s $video_path;

# Output the HTTP headers
print "Content-Type: video/mp4\n";
print "Content-Length: $video_size\n\n";

# Send the video file
open my $video_file, '<', $video_path or die "Cannot open video file: $!";
binmode $video_file;
print while <$video_file>;
close $video_file;

File: www/website/cgi-bin/video.php
<?php
$video_path = "/home/aismaili/Webserv/www/website/pics/NPC.mp4";

// Check if the video file exists
if (!file_exists($video_path)) {
    header("Content-Type: text/plain");
    echo "Error: Video file not found.";
    exit(1);
}

// Get the size of the video file
$video_size = filesize($video_path);

// Output the HTTP headers
echo("Content-Type: video/mp4\n");
header("Content-Length: " . $video_size);

// Send the video file
readfile($video_path);
?>